docs/DevelopmentGuide.rst

```diff
@@ -0,0 +1,105 @@
+
+2. Clone your fork locally
+
+    .. code-block:: console
+
+        $ git clone git@github.com:your_name_here/mqt-bench
+
+
+3. Change into the project directory
+
+    .. code-block:: console
+
+        $ cd mqt-bench
+
+4. Create a branch for local development
+
+    .. code-block:: console
+
+        $ git checkout -b name-of-your-bugfix-or-feature
+
+    Now you can make your changes locally.
+
+5. (Optional, **highly recommended**) Set up a virtual environment
+
+    .. code-block:: console
+
+        $ python3 -m venv venv
+        $ source venv/bin/activate
+
+    .. note::
+
+        If you are using Windows, you can use the following command instead:
+
+        .. code-block:: console
+
+            $ python3 -m venv venv
+            $ venv\Scripts\activate.bat
+
+    Ensure that pip, setuptools, and wheel are up to date:
+
+    .. code-block:: console
+
+        (venv) $ pip install --upgrade pip, setuptools, wheel
+
+
+6. (Optional) Install `pre-commit <https://pre-commit.com/>`_ to automatically run a set of checks before each commit.
+
+    .. code-block:: console
+
+        (venv) $ pipx install pre-commit
+        (venv) $ pre-commit install
+
+    If you use macOS, then pre-commit is in brew, use :code:`brew install pre-commit`.
+
+Building the Python module
+##########################
+
+The recommended way of building the Python module is to perform an editable install using `pip <https://pip.pypa.io/en/stable/>`_.
+
+    .. code-block:: console
+
+        (venv) $ pip install --editable .
+
+The :code:`--editable` flag ensures that changes in the Python code are instantly available without re-running the command.
+
+Running Python Tests
+--------------------
+
+The Python part of the code base is tested by unit tests using the `pytest <https://docs.pytest.org/en/latest/>`_ framework.
+The corresponding test files can be found in the :code:`tests/` directory.
+
+    .. code-block:: console
+
+        (venv) $ pytest
+
+This installs all dependencies for running the tests in an isolated environment, builds the Python package, and then runs the tests.
+The :code:`-r` flag ensures that the environment is reused for subsequent runs.
+
+Python Code Formatting and Linting
+----------------------------------
+
+The Python code is formatted and linted using a collection of `pre-commit hooks <https://pre-commit.com/>`_.
+This collection includes:
+
+- `black <https://docs.astral.sh/ruff/>`_ -- a code linter and formatter that automatically formats Python code according to the `PEP 8 style guide <https://www.python.org/dev/peps/pep-0008/>`_
+- `isort <https://pycqa.github.io/isort/>`_ -- a tool that automatically sorts Python imports according to the `PEP 8 style guide <https://www.python.org/dev/peps/pep-0008/>`_
+- `mypy <http://mypy-lang.org/>`_ -- a static type checker for Python code
+
+There are two ways of using these hooks:
```

Comment:
I only see one way being described down below ;o)

Reply:
Fixed it.